msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ManagedIdentityClient.java

```diff
@@ -38,6 +38,8 @@ private static AbstractManagedIdentitySource createManagedIdentitySource(MsalReq
             return managedIdentitySource;
         } else if ((managedIdentitySource = CloudShellManagedIdentitySource.create(msalRequest, serviceBundle)) != null) {
             return managedIdentitySource;
+        } else if ((managedIdentitySource = ServiceFabricManagedIdentitySource.create(msalRequest, serviceBundle)) != null) {
```

Comment:
In the .NET implementation, the order is different: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

Looks like `ServiceFabric` needs to be chcked first. I remember from discussion with Azure SDK that order if important.

Reply:
Right, move the service fabric to the top. Wondering if the unit tests are passing with this. If they are then there is something wrong.

Reply:
Fixed in the latest commit